### ISSUE TITLE
`smt run` with command line params allows passing of parameters not in param file

### ISSUE THREAD

**Issue by ashander** [open-research/sumatra#222]
From reading docs, this is unexpected as `smt help run` describes modifying 'default values'. 

To me, it's undesired, as it allows me to pass in params (by mistake, say) that my script doesn't use.

Seems natural to fix this along with Issue #221 (which involves param parsing).

To reproduce (in a virtual env)
```sh
$ pip sumatra
$ pip mercurial
$ mkdir smttest
$ cd smttest
$ hg init
$ smt init test-param
$ hg add main.py
$ hg commit -m "test param"
$ smt run -m main.py -e python default.param
__builtins__
seed
distr
n
...
$ smt run -m main.py -e python default.param b=3
__builtins__
distr
seed
b
n
...
```
with `default.param`
```sh
seed = 65785        # seed for random number generator
distr = "uniform"   # statistical distribution to draw values from
n = 100             # number of values to draw
```
and `main.py`
```python 

import sys

def run():
    parameter_file = sys.argv[1]
    parameters = {}
    execfile(parameter_file, parameters) # this way of reading parameters
                                         # is not necessarily recommended
     for p in parameters:
	print(p)
    data = [str(parameters["n"] * parameters["seed"]), ]

    f = file("Data/example2.dat", "w")
    f.writelines(str(data))
    f.close()

if __name__ == "__main__":
    run()
```

**Comment by ashander** [comment]
Addressed by #223 for all current parsers -- current behavior is to warn and pass

**Comment by apdavison** [comment]
I'm happy to have this raise an Exception rather than a Warning. It could be seen as a convenience for exploratory programming, but probably the case of passing in mis-named arguments by mistake will be the most common.
